pkg/packing/instance.go

```diff
@@ -0,0 +1,89 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package packing
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/awslabs/karpenter/pkg/utils/binpacking"
+	"github.com/awslabs/karpenter/pkg/utils/resources"
+	resourcesUtil "github.com/awslabs/karpenter/pkg/utils/resources"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type Instance struct {
+	// TODO replace w/ generic instance parameters
+	ec2.InstanceTypeInfo
+	Zones []string
+}
+
+type PackableInstance struct {
+	*Instance
+	reserved     v1.ResourceList
+	total        v1.ResourceList
+}
+
+func PackableInstancesFrom(instanceTypes []*Instance, overhead v1.ResourceList) []*PackableInstance {
```

Comment:
do these need to be exported? I don't think the cloud provider needs these funcs